packages/client-react-streaming/src/DataTransportAbstraction/WrapApolloProvider.tsx

```diff
@@ -59,18 +58,16 @@ export function WrapApolloProvider<ExtraProps>(
     ...extraProps
   }) => {
     const clientRef = useRef<ApolloClient<any>>(undefined);
-
-    if (process.env.REACT_ENV === "ssr") {
-      if (!clientRef.current) {
+    if (!clientRef.current) {
+      if (process.env.REACT_ENV === "ssr") {
         clientRef.current = makeClient();
+      } else {
+        clientRef.current = window[ApolloClientSingleton] ??= makeClient();
       }
-    } else {
-      clientRef.current = window[ApolloClientSingleton] ??= makeClient();
-    }
-
-    if (!(clientRef.current instanceof ApolloClient)) {
-      throw new Error(
-        `When using \`ApolloClient\` in streaming SSR, you must use the \`${WrappedApolloProvider.info.client}\` export provided by \`"${WrappedApolloProvider.info.pkg}"\`.`
+      assertInstance(
```

Comment:
No need to do this check on every render, doing it in the one render that initializes `clientRef` is enough.